contracts/cryptoblades.sol

```diff
@@ -296,17 +302,17 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
     }
 
     function fight(uint256 char, uint256 wep, uint32 target, uint8 fightMultiplier) external
-        onlyNonContract() {
+        onlyNonContract() returns (uint256) {
```

Comment:
the onlyNonContract modifier no longer makes sense here and users can bypass the fee by just calling this method directly. Should be a restricted check instead